src/pools/selection-strategies.ts

```diff
@@ -137,9 +137,9 @@ class DynamicPoolWorkerChoiceStrategy<Worker extends IWorker, Data, Response>
 
   /** @inheritdoc */
   public choose (): Worker {
-    const freeWorker = this.pool.findFreeTasksMapEntry()[0]
-    if (freeWorker) {
-      return freeWorker
+    const freeWorkerEntry = this.pool.findFreeTasksMapEntry()
+    if (freeWorkerEntry) {
+      return freeWorkerEntry[0]
```

Comment:
It's a freeTaskMapEntry. TS transpiler is Ok? Because I have had an error here while testing the same.

Reply:
Check it out and test it on your local setup
Maybe you just missed something when trying yourself, but now everything is just working (on my side)